components/style/properties/shorthand/position.mako.rs

```diff
@@ -138,6 +138,107 @@
 
 </%helpers:shorthand>
 
+% for kind in ["row", "column"]:
+<%helpers:shorthand name="grid-${kind}" sub_properties="grid-${kind}-start grid-${kind}-end"
+                    spec="https://drafts.csswg.org/css-grid/#propdef-grid-${kind}"
+                    products="gecko">
+    use values::specified::GridLine;
+    use parser::Parse;
+
+    pub fn parse_value(context: &ParserContext, input: &mut Parser) -> Result<Longhands, ()> {
+        let start = input.try(|i| GridLine::parse(context, i))?;
+        let end = if input.try(|i| i.expect_delim('/')).is_ok() {
+            GridLine::parse(context, input)?
+        } else {
+            let mut line = GridLine::default();
+            line.ident = start.ident.clone();       // ident from start value should be taken
```

Comment:
> if the first value is a \<custom-ident\>, the grid-row-end/grid-column-end longhand is also set to that \<custom-ident\>; otherwise, it is set to auto.

According to spec, the second ident should be set if start is `<custom-ident>` only. We should check that here.